MIDI/ambrosebs_MIDI Drum Trainer.jsfx

```diff
@@ -540,6 +542,40 @@ function setInputMaxVelocity(lane, value) local(idx)
   setSlider(lane, slider_offset_in_max_velocity, value, 0, 127, 0);
 );
 
+// Move a lane from oldPos to newPos, shifting other lanes to fill holes
+// and preserving the order of shifted lanes.
+function moveLane(oldPos, newPos) local(tmp, direction, currentPos, i, oldIdx, newIdx)
+(
+  (oldPos != newPos && oldPos >= 0 && newPos >= 0 && oldPos < lanes_limit && newPos < lanes_limit) ? (
+    direction = (newPos > oldPos) ? 1 : -1;
+    currentPos = oldPos;
+    
+    // Swap each slider one position at a time until we reach newPos
```

Comment:
Please elaborate in this comment how the algorithm progresses, in terms of the start/end positions and direction of the traversal.